### PR TITLE
UI Tester Module Reorganization

### PR DESCRIPTION
**Pull request by aaronayres35**
Part of #1248 

This PR pulls numerous modules and files into a new sub directory `_ui_tester_register` as specified by 1248.  The movement of modules is broken up to be one change per commit to hopefully make it a bit easier to review. 

After this I believe 1248 is very nearly finished, we just need to "Move _get_ui_registry from ui_tester.py to inside get_default_registry" in default_registry.  

~~Edit: actually I am just going to make that change in one last commit on this PR so that this PR can close up 1248~~ Scratch that I think it is better to put that in a smaller separate PR

### CODE REVIEW

File: traitsui/testing/tester/_ui_tester_register/default_registry.py
Comment (kitchoi) [inline traitsui/testing/tester/_ui_tester_register/default_registry.py:36]: Minor: This would work too, which would have been immune to module name change:
```suggestion
            ".".join([__name__, toolkit])
```

File: traitsui/testing/tester/_ui_tester_register/qt4/_interaction_helpers.py
Comment (kitchoi) [inline traitsui/testing/tester/_ui_tester_register/qt4/_interaction_helpers.py:14]: Darn... I think I had meant to type `_ui_tester_registry`. Sorry :(
The good thing about a name this unique is that is easy to do find-and-replace-all later. (This reminds me of one of the tips for an intermediate state of refactoring is to name something obviously bogus but also extremely unique so that it is easy to change them everywhere.)
The difference is really inconsequential, so I'd leave this up to you if you'd want to change this again.

**Review by kitchoi — APPROVED** [review]
LGTM, if CI is happy.

File: traitsui/testing/tester/_ui_tester_register/qt4/_interaction_helpers.py
Comment (aaronayres35) [reply traitsui/testing/tester/_ui_tester_register/qt4/_interaction_helpers.py:14]: I noticed that but then just proceeded anyway, I should've asked to confirm.  Like you said though this will be a simple find-and-replace-all, so I will do that in a quick separate PR

File: traitsui/testing/tester/_ui_tester_register/default_registry.py
Comment (aaronayres35) [reply traitsui/testing/tester/_ui_tester_register/default_registry.py]: The CI is failing with:
```
File "/Users/appveyor/.edm/envs/traitsui-test-3.6-pyqt/lib/python3.6/site-packages/traitsui/testing/tester/_ui_tester_register/default_registry.py", line 36, in get_default_registry
    ".".join([__name__, toolkit]))
  File "/Users/appveyor/.edm/envs/traitsui-test-3.6-pyqt/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 941, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 950, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'traitsui.testing.tester._ui_tester_register.default_registry.qt4'; 'traitsui.testing.tester._ui_tester_register.default_registry' is not a package
```
as __name__ is including `.default_registry` before toolkit.  There is probably an easy solution to this, but I think for now I will just undo this change and merge this PR.  That change can be made in a simple separate PR

Comment (kitchoi) [reply traitsui/testing/tester/_ui_tester_register/default_registry.py:36]: Right, that needs to be the package, not the module
```
this_package, _ = __name__.rsplit(".", 1)
```
Then the `this_package` should be `"traitsui.testing.tester._ui_tester_register"`.

Comment (kitchoi) [reply traitsui/testing/tester/_ui_tester_register/default_registry.py:36]: >  There is probably an easy solution to this, but I think for now I will just undo this change and merge this PR. That change can be made in a simple separate PR

SGTM.

**Pull request merged**
